core/src/ast_builder/select/join.rs

```diff
@@ -0,0 +1,543 @@
+use {
+    super::{NodeData, Prebuild},
+    crate::{
+        ast::{
+            Join, JoinConstraint, JoinExecutor, JoinOperator, ObjectName, Statement, TableAlias,
+            TableFactor,
+        },
+        ast_builder::{
+            ExprList, ExprNode, FilterNode, GroupByNode, JoinConstraintNode, LimitNode, OffsetNode,
+            ProjectNode, SelectItemList, SelectNode,
+        },
+        result::Result,
+    },
+};
+
+#[derive(Clone)]
+pub enum JoinType {
```

Comment:
1. For this kind of small enums, it would be good to dervice `Copy` trait.
Then, we don't need to use `clone()`.

e.g.
```rust
#[derive(Clone, Copy)]
pub enum JoinType { .. }
```

2. Name to `JoinOperatorType`.
There exists two different types which describes join.
One is join operator type which consists of inner or outer joins.
The other is join execution type, currently we support two join execution types - nested loop & hash joins.

So it would be better to use more explicit name for this.
This `JoinType` is for describing join operator types, then how about using `JoinOperatorType`?